File: src/main/java/dev/slimevr/autobone/AutoBoneListener.java
package dev.slimevr.autobone;

import java.util.EnumMap;

import dev.slimevr.autobone.AutoBone.Epoch;
import dev.slimevr.poserecorder.PoseFrames;
import dev.slimevr.vr.processor.skeleton.SkeletonConfigOffsets;


public interface AutoBoneListener {

	void onAutoBoneProcessStatus(
		AutoBoneProcessType processType,
		String message,
		long current,
		long total,
		boolean completed,
		boolean success
	);

	void onAutoBoneRecordingEnd(PoseFrames recording);

	void onAutoBoneEpoch(Epoch epoch);

	void onAutoBoneEnd(EnumMap<SkeletonConfigOffsets, Float> configValues);
}

File: server/core/src/main/java/dev/slimevr/autobone/AutoBoneListener.kt
package dev.slimevr.autobone

import dev.slimevr.autobone.AutoBone.Epoch
import dev.slimevr.poseframeformat.PoseFrames
import dev.slimevr.tracking.processor.config.SkeletonConfigOffsets
import java.util.*

interface AutoBoneListener {
	fun onAutoBoneProcessStatus(
		processType: AutoBoneProcessType,
		message: String?,
		current: Long,
		total: Long,
		eta: Float,
		completed: Boolean,
		success: Boolean,
	)

	fun onAutoBoneRecordingEnd(recording: PoseFrames)
	fun onAutoBoneEpoch(epoch: Epoch)
	fun onAutoBoneEnd(configValues: EnumMap<SkeletonConfigOffsets, Float>)
}
